### PR TITLE
A bunch of fixes

### PR DESCRIPTION
**Pull request by NiLuJe**
* VirtualKeyboardPopup stuff mentioned in #7696
* A bunch of PB stuff (#7732, #7747)

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7754)

### CODE REVIEW

**Comment by NiLuJe** [comment]
The PB startup script is, of course, completely untested.

Ping @Diniboy1123 (ideally, both an OTA and a crash ought to be tested. OTA might be annoying, so I'll settle for a crash: just send a SIGTERM to luajit, that'll do the trick).

File: frontend/ui/widget/virtualkeyboard.lua [hunk 1]
```diff
@@ -630,13 +641,19 @@ function VirtualKeyPopup:init()
     }
     if position_container.dimen.x < 0 then
         position_container.dimen.x = 0
+        -- We effectively move the popup, which means the key underneath our finger is no longer parent_key.
```
Comment (poire-z) [inline frontend/ui/widget/virtualkeyboard.lua:644]: `is no longer parent_key` => `may not be...` (offset may be a fraction of a key width, so we may or may not be the parent key). Still work in that case ?

Comment (NiLuJe) [reply frontend/ui/widget/virtualkeyboard.lua:644]: That's still an offset, so, technically, yeah.

Comment (poire-z) [reply frontend/ui/widget/virtualkeyboard.lua:644]: OK, first commit seems to work for me, thanks :)

**Review by Frenzie — APPROVED** [review]
Seems okay at a quick try.

**Comment by NiLuJe** [comment]
(Just a minor reword of the VK comment).

**Review by poire-z — APPROVED** [review]

**Comment by Diniboy1123** [comment]
@NiLuJe Sorry for the late reaction, my week has been stuffed until now. However I copied the new .app file to the device and it seems to start up well. Everything works. I can even exit the app fine now. Sending a sigterm to luajit indeed created a crashlog as well. I assume all works perfectly until now.

Thank you for this patch!

**Comment by NiLuJe** [comment]
@Diniboy1123: No worries ;).

For the record:

* Sending a SIGTERM to *luajit* should trigger something like this: https://github.com/koreader/koreader/pull/5328#issuecomment-528927190
(As well as generate some extra obvious markers in the crash.log).
* Sending a SIGTERM to *koreader.app* (or whatever the startup script ends up being shown as on PB) will do *nonne of the above* ^^.

Ideally, I would have you test the former, not the latter ;).

**Comment by Diniboy1123** [comment]
Yeah, I sent it to luajit since we wanted to see if the bash script logs the crash properly or not. Yet there is not much info regarding the crash logged in crash.log. Just the normal startup info, a bunch of .so files it couldn't load etc.

I couldn't see the active crash screen as I had to keep the app running but in the background and then send the signal using pbterm. But when I went back to koreader, the window was just empty white.

**Comment by NiLuJe** [comment]
No idea how the PB app management thingy affect this, but, yeah, I was assuming no weird PB crap like that, so, the signal being sent while the app is in the foreground (e.g., via SSH) ;).

Oh, well. At least it starts, so, eh.

**Comment by Diniboy1123** [comment]
I could probably compile an ssh binary or telnet if you would be interested and make it run somehow.

Anyway the app starts fine, works perfectly but when I would hit the exit from the menu (while a book is open) and then I choose exit koreader, it freezes the whole reader. I guess I will open a separate issue or just live with this.

**Comment by NiLuJe** [comment]
> Anyway the app starts fine, works perfectly but when I would hit the exit from the menu (while a book is open) and then I choose exit koreader, it freezes the whole reader. I guess I will open a separate issue or just live with this.

Is that specific to this change?

**Comment by Diniboy1123** [comment]
> Is that specific to this change?

Before the change it didn't even run, so no clue. But I don't think it should be connected. I will open a separate issue later if I know more why this happens.

**Pull request merged**